docs/developer/writing-flow-component-documentation.md

```diff
@@ -0,0 +1,512 @@
+block](#arguments).
+
+Values for the description column should be kept as short as possible to keep
+the table small and readable.
+
+A paragraph with the content "The following fields are exported and can be
+referenced by other components:" should always prefix the exported fields
+table.
+
+Following the exported fields table, a longer description of each exported
+fields may be provided, but this normally isn't done.
+
+### Component health
+
+The Component health section describes when the component is healthy. The
+section starts with an `h2` header called Component health.
+
+If the component does not have special health logic, the content of the section
+should be the following paragraph:
+
+```markdown
+`COMPONENT_NAME` is only reported as unhealthy if given an invalid
+configuration.
+```
+
+Otherwise, write a detailed description for when the component is reported as
+healthy.
+
+### Debug information
+
+The Debug information section describes debug information exposed in the
+Grafana Agent Flow UI. The section starts with an `h2` header called Debug
+information.
+
+If the component does not expose any debug information, the content of the
+section should be the following paragraph:
+
+```markdown
+`COMPONENT_NAME` does not expose any component-specific debug information.
+```
+
+Otherwise, write a high-level description for what debug information the
+component provides. Do not document the attributes or blocks which is exposed
+through the debug information.
+
+### Debug metrics
+
+The Debug metrics section describes what Prometheus metrics are exposed by a
+component. The section starts with an `h2` header called Debug metrics.
+
+If the component does not expose any debug metrics, the content of the section
+should be the following paragraph:
+
+```markdown
+`COMPONENT_NAME` does not expose any component-specific debug metrics.
+```
+
+When the component exports values, it should provide a table of exposed metrics
+with the following columns:
+
+| Column      | Description                       |
+| ----------- | --------------------------------- |
+| Name        | Name of Prometheus metric.        |
+| Type        | Prometheus metric type.           |
+| Description | Metric description.               |
+
+Values in the Name and Type column should be backticked.
+
+Values in the Type column should be a Prometheus metric type, one of `counter`,
+`gauge`, `histogram`, `native histogram`, or `summary`.
+
+A paragraph with the content "The following Prometheus metrics are exposed:"
+should always prefix the metrics table.
+
+### Examples
+
+The Examples section provides copy-and-paste River pipelines which use the
+River component. The section starts with an `h2` header called Examples. If
+there is only one example, call the section Example instead.
+
+The Examples section should be composed of a brief description of each example,
+followed by the example in a code block. For example:
+
```

Comment:
It might be nice to add if we support adding clarifying comments inside the examples or not.

For example, do we want something like:
```
prometheus.exporter.example {
    // The exclude list will always be prioritized over the include list, so "db2" will not appear in
    // exported metrics.
    include = ["db1", "db2"]
    exclude = ["db2"]
}

Reply:
hmm, to me it feels like that kind of information should be put in the Arguments section (or the relevant Block section) instead of having people search for an example

Reply:
That's fair, maybe it would be good in certain circumstances to have it in both? That would mean that the information is available in Arguments/Block section, but with a concrete example to reinforce understanding as well?